receiver/dcgmreceiver/client.go

```diff
@@ -113,7 +113,7 @@ func newClient(settings *dcgmClientSettings, logger *zap.Logger) (*dcgmClient, e
 		deviceGroup:                    deviceGroup,
 		devices:                        map[uint]deviceMetrics{},
 		lastSuccessfulPoll:             time.Now(),
-		deviceMetricToFailedQueryCount: make(map[string]uint64),
+		deviceMetricToFailedQueryCount: make(map[string]int),
```

Comment:
Just curious, was this so you wouldn't have to declare `limit` as `uint64` as well?

Reply:
That's why I noticed, yeah, but I don't see any reason this needed to be `uint64` - in general in Go, anything that's counting should just use `int` (basically the equivalent of `size_t` in C).